prov/tcp/src/xnet_init.c

```diff
@@ -132,6 +131,12 @@ static void xnet_init_env(void)
 			"applications to prevent hangs. (default: %d)",
 			xnet_max_saved);
 	fi_param_get_int(&xnet_prov, "max_saved", &xnet_max_saved);
+
+	fi_param_define(&xnet_prov, "max_rx_size", FI_PARAM_SIZE_T,
+			"maximum size for message buffers at the receiver "
+			"(default: %zu)", xnet_max_rx_size);
+	fi_param_get_size_t(&xnet_prov, "max_rx_size", &xnet_max_rx_size);
```

Comment:
We need to check that the size >= max_inject and increase if not.  I'd leave the name for the environment variable as-is (max_rx_size), as it's how a user would view it.